packages/tpetra/core/src/Tpetra_Details_Behavior.cpp

```diff
@@ -22,6 +24,89 @@ namespace { // (anonymous)
     return s;
   }
 
+  template<class Func>
+  void
+  split(const std::string& s, Func f, const char sep=':')
```

Comment:
You could hide the implementation of `split`, by using `std::function<void(const std::string&)>` as the type of `f`, instead of templating on `Func`.  This would also better express intent, without restricting the interface in any way (the code should be equivalent).

Reply:
I'm getting super into implementation hiding as a result of working with @prwolfe :-) .

Reply:
Another thing: If `:` is the separator, then what do we do about classes in the `Tpetra::Details` namespace?  I'm OK with just ignoring them, but just want to know what you think.

Reply:
I chose `:` mainly for its use as the posix pathsep character.  The name passed to `verbose` or `debug` doesn't necessarily have to be fully namespace qualified.